rubber_duck.py

```diff
@@ -133,12 +139,12 @@ async def have_conversation(self, thread_id: int, engine: str, prompt: str, init
                     return
 
     @step
-    async def _get_completion(self, thread_id, engine, message_history) -> tuple[list, dict]:
+    async def _get_completion(self, thread_id, engine, message_history) -> dict:
         # Replaces _get_response
         async with self._typing(thread_id):
-            completion: OpenAIObject = await openai.ChatCompletion.acreate(
+            completion: ChatCompletion = await client.chat.completions.create(
                 model=engine,
                 messages=message_history
             )
             logging.debug(f"Completion: {completion}")
-            return completion.choices, completion.usage
+            return completion.dict()
```

Comment:
return the choices and usage dictionaries like we did before. Those sub objects should also have a .dict() method. Then the return type and calling code doesn't need to be different.

Reply:
I can do that for usage, but not for choices. `completion.choices` returns a list, not a dictionary. I'll fiddle around with it and see what I can do
[completion.json](https://github.com/beanlab/rubber-duck/files/14254728/completion.json)

Reply:
Ok, I was able to parse out the list of choices and usage as a dictionary before returning and that worked when I ran it on my end